src/checkers/flashchecker.py

```diff
@@ -0,0 +1,130 @@
+# Flash Checker: A checker that to see if the url is pointing to the latest Flash Player.
+#
+# The contens of the x-checker-data for the module should be .e.g:
+#   "x-checker-data": {
+#                       "type": "flash",
+#                       "browser": "Chrome|Firefox",
+#                     }
+#
+# In addition, you *MUST* set only-arches on the source itself, which the checker will use
+# to determine which architecture to check.
+#
+# Copyright © 2019 Endless Mobile, Inc.
+#
+# Authors:
+#       Ryan Gonzalez <rymg19@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import json
+import logging
+import urllib.error
+import urllib.request
+import urllib.parse
+
+from lib.externaldata import ExternalData, ExternalFile, Checker
+from lib import utils
+
+log = logging.getLogger(__name__)
+
+
+FLASH_BASE_URL = 'http://get.adobe.com/flashplayer/webservices/json/?{params}'
+
+FLATPAK_TO_FLASH_ARCH_MAP = {
+    'i386': 'x86-32',
+    'x86_64': 'x86-64',
+}
+
+BROWSER_TO_PAPI_MAP = {
```

Comment:
Why not just specify `ppapi`/`npapi` in the checker data?

Reply:
Later in the code we have need the browser name anyway to find the proper URL, and having to specify both the browser *and* the plugin API feels redundant when one is always associated with the other.

I mean it could be the other way around (specify the plugin API and derive the browser name), but using the browser name felt clearer.